EzMap.Domain/Repositories/PoiRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Runtime.InteropServices.ComTypes;
+using EzMap.Domain.Dtos;
+using EzMap.Domain.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace EzMap.Domain.Repositories;
+
+public interface IPoiRepository
+{
+    Task<bool> AddPoi(PoiCreateDto dto);
+
+    Task ReadBooksAsync(CancellationToken token = default);
+}
+
+
+public class PoiRepository : IPoiRepository
```

Comment:
All repos should filter out "soft-deleted" entities as well.
We can do this in repo logic or use Ef's [Global query filter](https://learn.microsoft.com/en-us/ef/core/querying/filters)